src/beamlime/handlers/config_handler.py

```diff
@@ -110,40 +111,48 @@ def handle(self, messages: list[Message[RawConfigItem]]) -> list[Message[None]]:
         :
             Empty list as this handler doesn't produce output messages
         """
-        updated: dict[str, ConfigUpdate] = {}
+        # Stores the most recent update for this key/source combination
+        updated: defaultdict[str, dict[str | None, ConfigUpdate]] = defaultdict(dict)
         for message in messages:
             try:
                 update = ConfigUpdate.from_raw(message.value)
                 if update.service_name not in (None, self._service_name):
                     # Ignore messages not for this service
                     continue
                 source_name = update.source_name
-                key = update.key
+                config_key = update.key
                 value = update.value
                 self._logger.info(
                     'Updating config for source_name = %s: %s = %s at %s',
                     source_name,
-                    key,
+                    config_key,
                     value,
                     message.timestamp,
                 )
-                updated[key] = update
                 if source_name is None:
-                    self._global_store[key] = value
-                    for store in self._stores.values():
-                        store[key] = value
+                    self._global_store[config_key] = value
+                    for source_name, store in self._stores.items():
+                        updated[config_key][source_name] = update
+                        store[config_key] = value
                 else:
-                    self.get_config(source_name)[key] = value
+                    updated[config_key][source_name] = update
+                    self.get_config(source_name)[config_key] = value
```

Comment:
This line can also be done after collecting all updated configurations.

It doesn't matter so much, but we might want to watch this configuration object to cache it or sth... in the future....?
But feel free to ignore this comment.